Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -797,57 +932,86 @@ void MaxEnt::populateImageWS(MatrixWorkspace_const_sptr &inWS, size_t spec,
   outWS->setSharedE(nspec + spec, outWS->sharedE(spec));
 }
 
-/** Populates the output workspaces
+/** Populates the data output workspace
 * @param inWS :: [input] The input workspace
 * @param spec :: [input] The current spectrum being analyzed
 * @param nspec :: [input] The total number of histograms in the input workspace
 * @param result :: [input] The reconstructed data to be written in the output
 * workspace (can be a real or complex vector)
 * @param complex :: [input] True if result is a complex vector, false otherwise
+* @param concatenated :: [input] True if result is concatenated spectra,
+* then all spectra are analyzed and spec must be 0.
 * @param outWS :: [input] The output workspace to populate
 */
 void MaxEnt::populateDataWS(MatrixWorkspace_const_sptr &inWS, size_t spec,
                             size_t nspec, const std::vector<double> &result,
-                            bool complex, MatrixWorkspace_sptr &outWS) {
+                            bool concatenated, bool complex,
+                            MatrixWorkspace_sptr &outWS) {
 
   if (complex && result.size() % 2)
-    throw std::invalid_argument("Cannot write results to output workspaces");
-
-  int npoints = complex ? static_cast<int>(result.size() / 2)
-                        : static_cast<int>(result.size());
-  int npointsX = inWS->isHistogramData() ? npoints + 1 : npoints;
-  MantidVec X(npointsX);
-  MantidVec YR(npoints);
-  MantidVec YI(npoints);
-  MantidVec E(npoints, 0.);
-
+    throw std::invalid_argument(
+        "Cannot write data results to output workspaces");
+  if (concatenated && !complex)
+    throw std::invalid_argument("Concatenated data results must be complex");
+  if (concatenated && result.size() % nspec * 2)
+    throw std::invalid_argument(
+        "Cannot write complex concatenated data results to output workspaces");
+  if (concatenated && spec != 0)
+    throw std::invalid_argument("Cannot write concatenated data results to "
+                                "output workspaces from non-first spectrum");
+
+  int resultLength = complex ? static_cast<int>(result.size() / 2)
+                             : static_cast<int>(result.size());
+  size_t spectrumLength = (concatenated ? resultLength / nspec : resultLength);
+  size_t spectrumLengthX =
+      inWS->isHistogramData() ? spectrumLength + 1 : spectrumLength;
+  size_t nSpecAnalyzed = (concatenated ? nspec : 1);
+
+  // Here we assume equal constant binning for all spectra analyzed
   double x0 = inWS->x(spec)[0];
   double dx = inWS->x(spec)[1] - x0;
 
-  // X values
-  for (int i = 0; i < npointsX; i++) {
-    X[i] = x0 + i * dx;
-  }
+  // Loop over each spectrum being analyzed - one spectrum unless concatenated
+  for (size_t specA = spec; specA < spec + nSpecAnalyzed; specA++) {
 
-  // Y values
-  if (complex) {
-    for (int i = 0; i < npoints; i++) {
-      YR[i] = result[2 * i];
-      YI[i] = result[2 * i + 1];
+    MantidVec X(spectrumLengthX);
```

Comment:
should these not be the ```binEdges``` and ```counts``` or ```HistogramX``` etc. data types? http://docs.mantidproject.org/v3.9.2/concepts/HistogramData.html

Reply:
I haven't seen these data types used here before. I was just using the data types already in use.

Reply:
I've created issue #23106 to deal with this.